### DISCUSSION TITLE
Release 0.43 is out and adds Home Assistant MQTT Discovery!

### DISCUSSION THREAD

**Discussion by Resinchem** [Resinchem/Tailwind2MQTT#3] [Announcements]
Release 0.43 is now out and adds an option to enable Home Assistant MQTT Discovery.  When enabled, it is no longer necessary to manually create sensors via YAML in your Home Assistant configuration file.  Instead, all sensors are discovered and added automatically for you!

This does require a change to your Settings. h file to add the new option.  There are two ways to approach this:

- Download the new Settings.h file as part of the release and update the new file with your original settings and options
- Simply add the following line to the end of your existing Settings.h file:  ```bool ha_discovery = true;```

If you opt for the second option, do not download or overwrite your existing Settings.h file.  You only need to snag the new tailwind_mqtt.ino file, compile and upload OTA to take advantage of the new feature.  When the board reboots, the sensors will magically appear in Home Assistant!

Note: This release does require one additional library ```ArduinoJSON.h```, so you may also need to install that library in your environment if it is missing.

See the wiki page [MQTT \ Home Assistant](https://github.com/Resinchem/Tailwind2MQTT/wiki/MQTT-%5C-Home-Assistant) for full details on the sensors that will be created and more!

Update:  I'm working on adding switches to the Home Assistant MQTT Discovery process, so that YAML will not be needed at all to both control the door(s) and get the status.  This is a little trickier than the sensors (due to the way HA handles payloads on MQTT switches), so it may be a bit before this is available.

**Comment by focher** [comment]
There's a typo in the reference to the `ArduinoJSON.h` package above.

**Reply by Resinchem** [reply]
Thanks...  updated!
